### PR TITLE
Merge the feature/embedding branch into master

### PR DESCRIPTION
**Pull request by hunger**
The `ComponentContainer` element and the `component-factory` type are all private, experimental for now. Let's try them out before we stabilize them!

### CODE REVIEW

File: internal/interpreter/api.rs [hunk 1]
```diff
@@ -42,6 +43,8 @@ pub enum ValueType {
     Brush,
     /// Correspond to `image` type in .slint.
     Image,
+    /// Correspond to `component-factory` type in .slint.
+    ComponentFactory,
```
Comment (ogoffart) [inline internal/interpreter/api.rs:47]: We shouldn't expose this in the public API 

```suggestion
```

File: internal/interpreter/api.rs [hunk 2]
```diff
@@ -65,6 +68,7 @@ impl From<LangType> for ValueType {
             LangType::Struct { .. } => Self::Struct,
             LangType::Void => Self::Void,
             LangType::Image => Self::Image,
+            LangType::ComponentFactory => Self::ComponentFactory,
```
Comment (ogoffart) [inline internal/interpreter/api.rs:71]:
```suggestion
```

File: internal/interpreter/api.rs [hunk 3]
```diff
@@ -115,6 +119,8 @@ pub enum Value {
     EnumerationValue(String, String) = 10,
     #[doc(hidden)]
     LayoutCache(SharedVector<f32>) = 11,
+    /// Correspond to the `component` type in .slint
+    ComponentFactory(ComponentFactory) = 12,
```
Comment (ogoffart) [inline internal/interpreter/api.rs:123]:
```suggestion
    #[doc(hidden)]
    /// Correspond to the `component-factory` type in .slint
    ComponentFactory(ComponentFactory) = 12,
```

File: api/rs/slint/lib.rs [hunk 4]
```diff
@@ -171,23 +171,23 @@ The follow table summarizes the entire mapping:
...
-| `duration` | `i64` | At run-time, durations are always represented as signed 64-bit integers with millisecond precision. |
-| `angle` | `f32` | The value in degrees |
+| `physical_length` | `f32` | The unit are physical pixels. |
+| `Point` | [`LogicalPosition`] | A struct with `x` and `y` fields, representing logical coordinates. |
 | `relative-font-size` | `f32` | Relative font size factor that is multiplied with the `Window.default-font-size` and can be converted to a `length`. |
+| `string` | [`SharedString`] | A reference-counted string type that can be easily converted to a str reference. |
 | structure | `struct` of the same name | |
-| anonymous object | anonymous tuple | The fields are in alphabetical order. |
-| enumeration | `enum` of the same name | The values are converted to CamelCase |
-| array | [`ModelRc`] | Arrays are represented as models, so that their contents can change dynamically. |
```
Comment (ogoffart) [inline api/rs/slint/lib.rs:189]: These are not  literal value and it makes sense to keep them separate at the end IMHO

File: docs/language/src/reference/types.md [hunk 5]
```diff
@@ -4,20 +4,21 @@ All properties in Slint have a type. Slint knows these basic types:
...
-| `length`             | The type used for `x`, `y`, `width` and `height` coordinates. Corresponds to a literal like `1px`, `1pt`, `1in`, `1mm`, or `1cm`. It can be converted to and from length provided the binding is run in a context where there is an access to the device pixel ratio.                                                                            | 0px           |
+| `color`              | RGB color with an alpha channel, with 8 bit precision for each channel. CSS color names as well as the hexadecimal color encodings are supported, such as `#RRGGBBAA` or `#RGB`.                                                                                                                                                                 | transparent   |
+| `component-factory`  | A factory object that can create components.                                                                                                                                                                                                                                                                                                     | None factory  |
```
Comment (tronical) [inline docs/language/src/reference/types.md:11]: This is documented as public type when it's not.

File: internal/core/items/component_container.rs [hunk 6]
```diff
@@ -0,0 +1,245 @@
...
+    }
+
+    fn layout_info(
+        self: Pin<&Self>,
+        orientation: Orientation,
+        _window_adapter: &Rc<dyn WindowAdapter>,
+    ) -> LayoutInfo {
+        self.ensure_updated();
+
+        // Query the component_factory property to force a re-layout when that changes
```
Comment (tronical) [inline internal/core/items/component_container.rs:186]: This comments seems either outdated or there's a dependency missing here.

**Pull request merged**